build.sh

```diff
@@ -29,6 +29,8 @@ PRINT_USAGE() {
     echo "  -j [N], --jobs[=N]   Multicore build, allow N jobs at once"
     echo "  -n, --ninja          Build with ninja instead of make"
     echo "      --xcode          Generate XCode project"
+    echo "      --gen[=Gen]      Give a custom generator to cmake"
+    echo "      --make[=cmd]     Give a custom make command"
```

Comment:
I wanted to use another cmake generator and it felt silly to just keep adding options for every generator.
Therefore, I figured it would be simpler to give the option to pass in anything you want.
in my case, I wanted CodeBlocks project file so I ran `./build.sh --gen=\"CodeBlocks - Unix Makefiles\""`

For the `--make` option, I saw that the make option is currently tied to the generator.
I didn't need it for my case as my generator creates project files, but I could easily see someone needing it for a different generator.